### PR TITLE
Feature/M2-1346 activity flow builder actions buttons, drag and drop

### PR DESCRIPTION
**Pull request by IhorNesterenko99**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ document.body.appendChild(svgBuilder(svgSprite));
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
 root.render(
   <React.StrictMode>
-    <App />
+    <App />,
```
Comment (alex-i-e) [inline src/index.tsx:16]: typo?

Comment (IhorNesterenko99) [reply src/index.tsx]: fixed

**Review by muhurov — APPROVED** [review]

File: src/shared/components/Actions/Actions.tsx [hunk 2]
```diff
@@ -16,55 +17,67 @@ export const Actions = ({
...
-                  </span>
-                </Tooltip>
-              );
-            },
-          )}
-        </StyledActions>
-      )}
-      {!isVisible && <StyledDotsSvg id="dots" width={18} height={4} />}
+            return (
+              <Tooltip key={uniqueId()} tooltipTitle={tooltipTitle}>
```
Comment (kdanilenka32) [inline src/shared/components/Actions/Actions.tsx:53]: pls add key={icon.props.id}

File: src/shared/components/Actions/Actions.tsx [hunk 3]
```diff
@@ -16,55 +17,67 @@ export const Actions = ({
...
+                  </StyledActionButton>
+                </span>
+              </Tooltip>
+            );
+          },
+        )}
+        {dragHandleProps && (
+          <StyledActionButton
+            isVisible={isVisible}
+            key={uniqueId()}
```
Comment (kdanilenka32) [inline src/shared/components/Actions/Actions.tsx:71]: pls replace uniqueId here

File: src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx [hunk 4]
```diff
@@ -125,38 +139,51 @@ export const ActivityFlow = () => {
...
-            </Fragment>
-          ))}
+        <DragDropContext onDragStart={() => setIsDragging(true)} onDragEnd={handleDragEnd}>
+          <DndDroppable droppableId="activity-flows-dnd" direction="vertical">
+            {(listProvided) => (
+              <Box {...listProvided.droppableProps} {...{ ref: listProvided.innerRef }}>
+                {activityFlows.map((item, index) => (
+                  <Draggable key={item.id} draggableId={item.id || ''} index={index}>
+                    {(itemProvided, snapshot) => (
+                      <Box {...itemProvided.draggableProps} {...{ ref: itemProvided.innerRef }}>
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx:149]:
```suggestion
                      <Box {...itemProvided.draggableProps} ref={itemProvided.innerRef}>
```

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 5]
```diff
@@ -32,18 +44,52 @@ export const ActivityFlowBuilder = () => {
...
+    setFlowActivityToDeleteData(null);
+  };
+
+  const handleFlowActivityAdd = (activityKey: string) =>
+    append({ id: uuidv4(), activityId: activityKey });
+
+  const handleFlowActivityToUpdateSet = (event: MouseEvent<HTMLElement>, index: number) => {
+    setIndexToUpdate(index);
+    let parentElement = event.currentTarget.parentNode as HTMLElement;
+    while (parentElement && !parentElement.classList.contains('builder-item')) {
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:68]: could you move `builder-item` in a separate const file?

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 6]
```diff
@@ -54,26 +100,95 @@ export const ActivityFlowBuilder = () => {
...
+        onClearFlow: remove,
+      }}
+    >
+      <DragDropContext onDragEnd={handleDragEnd}>
+        <DndDroppable droppableId="activity-flow-builder-dnd" direction="vertical">
+          {(listProvided) => (
+            <Box {...listProvided.droppableProps} {...{ ref: listProvided.innerRef }}>
+              {activityFlowItems?.map((item, index) => {
+                const activityKey = item.activityId;
+                const currentActivity = activities?.find(
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:118]: pls reduce complexity const Ids =activities.reduce(..., {})

const currentActivity = Ids[activityKey]

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 7]
```diff
@@ -54,26 +100,95 @@ export const ActivityFlowBuilder = () => {
...
+                const activityName = currentActivity?.name;
+                const activityDescription = currentActivity?.description;
+
+                return (
+                  <Draggable key={item.id} draggableId={item.id || ''} index={index}>
+                    {(itemProvided, snapshot) => (
+                      <Box
+                        className="builder-item"
+                        {...itemProvided.draggableProps}
+                        {...{ ref: itemProvided.innerRef }}
```
Comment (alex-i-e) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:130]:
```suggestion
                        ref={itemProvided.innerRef}
```

Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:130]: done

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx (hunk 5 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:68]: done

File: src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx [hunk 8]
```diff
@@ -125,38 +139,51 @@ export const ActivityFlow = () => {
...
-                    <Svg id="add" width={18} height={18} />
-                  </StyledAdd>
-                </StyledAddWrapper>
-              )}
-            </Fragment>
-          ))}
+        <DragDropContext onDragStart={() => setIsDragging(true)} onDragEnd={handleDragEnd}>
+          <DndDroppable droppableId="activity-flows-dnd" direction="vertical">
+            {(listProvided) => (
+              <Box {...listProvided.droppableProps} {...{ ref: listProvided.innerRef }}>
```
[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx:145]: done

File: src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx (hunk 4 above)
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlow/ActivityFlow.tsx:149]: done

File: src/shared/components/Actions/Actions.tsx (hunk 3 above)
Comment (IhorNesterenko99) [reply src/shared/components/Actions/Actions.tsx:71]: done

**Review by alex-i-e — APPROVED** [review]

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilderHeader/ActivityFlowBuilderHeader.tsx [hunk 9]
```diff
@@ -1,21 +1,76 @@
...
-import { StyledHeader, BuilderContainerProps } from 'shared/features';
+import { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@mui/material';
 
+import { Svg, ButtonWithMenu } from 'shared/components';
+import { StyledHeader } from 'shared/features';
+import { falseReturnFunc } from 'shared/utils';
+import { AppletFormValues } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.types';
```
[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilderHeader/ActivityFlowBuilderHeader.tsx:9]: done

File: src/shared/components/Actions/Actions.tsx (hunk 2 above)
Comment (IhorNesterenko99) [reply src/shared/components/Actions/Actions.tsx]: done

File: src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx [hunk 10]
```diff
@@ -54,26 +102,102 @@ export const ActivityFlowBuilder = () => {
...
+        onClearFlow: remove,
+      }}
+    >
+      <DragDropContext onDragEnd={handleDragEnd}>
+        <DndDroppable droppableId="activity-flow-builder-dnd" direction="vertical">
+          {(listProvided) => (
+            <Box {...listProvided.droppableProps} ref={listProvided.innerRef}>
+              {activityFlowItems?.map((item, index) => {
+                const activityKey = item.activityId;
+                const ids = activities.reduce(
```
Comment (kdanilenka32) [inline src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx:120]: move this ids outside of map and rename it to activitiesIds

**Pull request merged**